#!/usr/bin/env python

#package name
PACKAGE = "era_gazebo"

#imports
from dynamic_reconfigure.parameter_generator_catkin import *

#calling paramgen function
gen = ParameterGenerator()


#list of params
#in our case: single param -> mask threshold (between 0 to 1) to get the point cloud
#function gen.add()
#arguments: name, type, 0, description, default value, min, max
gen.add("mask_threshold",    double_t,    0, "Mask threshold value as parameter", 0.5,  0, 1)
gen.add("Median_Filter_type",   int_t,    0, "Median Filter Type: Standard: 0 and custom: 1 as parameter", 0,  0, 1)
#exit the generate function
gen.add("Median_Filter_kernel",   int_t,    0, "Median Filter Kernel size as parameter", 5,  3, 41)
#exit the generate function
#arguments: package, our package name, our config name
exit(gen.generate(PACKAGE, "era_gazebo", "test"))
